backend/src/main/java/pl/edu/pw/elka/etherscan/EtherscanFacade.java

```diff
@@ -11,7 +11,10 @@ public EtherscanFacade(EtherscanConnector etherscanConnector) {
     }
 
     public EtherscanTransactionsDto getTransactionsForAddress(String address) {
-        // TODO address validation
-        return etherscanConnector.getTransactions(address);
+        EthereumAddressValidator ethereumAddressValidator = new EthereumAddressValidator();
+            if (ethereumAddressValidator.checkAddres(address))
+                return etherscanConnector.getTransactions(address);
+            else
+                throw new BadEthereumAddressException("Wrong Ethereum address");
```

Comment:
tak jak pisalem - rzucenie wyjatku to bardziej odpowiedzialnosc validatora w tym przypadku